index.js

```diff
@@ -54,13 +56,76 @@ function decodeAssetCacheName(encodedString) {
   return decodedFileName.split(":");
 }
 
+function touch(path) {
+  const time = new Date();
+  try {
+    fs.utimesSync(path, time, time);
+  } catch (err) {
+    fs.closeSync(fs.openSync(path, "w"));
+  }
+}
+
+function evictLeastRecentlyUsed(cacheDir, maxSize, logger) {
+  getFolderSize(cacheDir, (err, size) => {
+    if (err) {
+      throw err;
+    }
+
+    if (size >= maxSize) {
+      try {
+        // find least recently used file
+        const leastRecentlyUsed = findLeastRecentlyUsed(cacheDir);
+
+        // and delete it
+        const { dir } = path.parse(leastRecentlyUsed.path);
+        fs.unlinkSync(leastRecentlyUsed.path);
+        fs.rmdirSync(dir);
+
+        if (logger) {
+          logger.info(`Evicted ${leastRecentlyUsed.path} from cache`);
+        }
+
+        evictLeastRecentlyUsed(cacheDir, maxSize);
```

Comment:
shouldn't the logger also be passed here?

Reply:
totally. fixed.